metricflow-semantics/tests_metricflow_semantics/snapshots/test_linkable_spec_resolver.py/str/test_create_linkable_element_set_from_join_path__result0.txt

```diff
@@ -1,126 +1,126 @@
+listings_latest   ("('listing', 'user')", "('user',)")                     lux_listings                                                                          ['JOINED', 'METRIC', 'MULTI_HOP']
+listings_latest   ("('listing', 'user')", "('user',)")                     regional_starting_balance_ratios                                                      ['JOINED', 'METRIC', 'MULTI_HOP']
+listings_latest   ("('listing', 'user')", "('user',)")                     revenue                                                                               ['JOINED', 'METRIC', 'MULTI_HOP']
+listings_latest   ("('listing', 'user')", "('user',)")                     revenue_all_time                                                                      ['JOINED', 'METRIC', 'MULTI_HOP']
+listings_latest   ("('listing', 'user')", "('user',)")                     smallest_listing                                                                      ['JOINED', 'METRIC', 'MULTI_HOP']
+listings_latest   ("('listing', 'user')", "('user',)")                     total_account_balance_first_day                                                       ['JOINED', 'METRIC', 'MULTI_HOP']
+listings_latest   ("('listing', 'user')", "('user',)")                     views                                                                                 ['JOINED', 'METRIC', 'MULTI_HOP']
+listings_latest   ("('listing', 'user')", "('user',)")                     visit_buy_conversion_rate                                                             ['JOINED', 'METRIC', 'MULTI_HOP']
+listings_latest   ("('listing', 'user')", "('user',)")                     visit_buy_conversion_rate_7days                                                       ['JOINED', 'METRIC', 'MULTI_HOP']
+listings_latest   ("('listing', 'user')", "('user',)")                     visit_buy_conversion_rate_7days_fill_nulls_with_0                                     ['JOINED', 'METRIC', 'MULTI_HOP']
+listings_latest   ("('listing', 'user')", "('user',)")                     visit_buy_conversion_rate_by_session                                                  ['JOINED', 'METRIC', 'MULTI_HOP']
+listings_latest   ("('listing', 'user')", "('user',)")                     visit_buy_conversions                                                                 ['JOINED', 'METRIC', 'MULTI_HOP']
+listings_latest   ("('listing', 'user')", "('verification', 'user')")      identity_verifications                                                                ['JOINED', 'METRIC', 'MULTI_HOP']
+listings_latest   ("('listing', 'user')", "('view', 'user')")              views                                                                                 ['JOINED', 'METRIC', 'MULTI_HOP']
+listings_latest   ("('listing', 'user')", "('visit', 'user')")             visit_buy_conversion_rate                                                             ['JOINED', 'METRIC', 'MULTI_HOP']
```

Comment:
I made my browser bigger so this is a little more obvious to me, but I still don't understand what's happening here. Is the second tuple a different internal join path for resolving the visit_buy_conversions metric? But then isn't that an ambiguous join? How do we know which one to use?

Reply:
The first tuple is the entity links in the OUTER query (how we join to the metric subquery).
The second tuple is the entity links in the INNER query (how the metric joins to the entity). There can be multiple options for this path, even when the outer path is the same.
That does make the second one ambiguous, but can be deduped by what they pass into the filter! Currently they would need to specify the entity links via dundered syntax, but I have a task on my list to support the `Entity(...)` so that they can specify the entity links without the dundered syntax.

Not sure if this actually addresses your confusion so let me know if not!

Reply:
OH I GET IT. This is the set of all metrics and their (internal) join paths, because from the outside those look like different metrics, and the user would have to indicate precisely which of these targets they want. So this is what we want the snapshot to represent. 

I wonder if we should make a separate column for this so we can label it, like internal_join_links or something, but I think this will do for now. No sense rewriting all the things right now, especially since that column can't be populated for anything other than a group by metric.